projet/users/models.py

```diff
@@ -74,6 +74,38 @@ class Profile(models.Model):
     bio = models.TextField(null=True, blank=True)
 
 
+class Contributor(models.Model):
+    user = models.ForeignKey(
+        User,
+        blank=False,
+        null=False,
+        on_delete=models.PROTECT,
+        related_name="contributions",
+    )
+    project = models.ForeignKey(
+        "projects.Project",
+        on_delete=models.CASCADE,
+        related_name="contributors",
+        unique=False,
+    )
+    accepted = models.BooleanField(default=False)
+    accepted_by = models.ForeignKey(
+        User, on_delete=models.PROTECT, blank=True, null=True
+    )
+    date_accepted = models.DateTimeField(blank=True, null=True)
+    is_admin = models.BooleanField(default=False)
+
+    def get_profile(self):
+        return self.user.profile
+
+    def save(self, *args, **kwargs):
+        if self.user == self.project.user:
+            return ValidationError(
+                "You are the manager and cannot be a contributor to this project"
+            )
+        super(Contributor, self).save(*args, **kwargs)
```

Comment:
A contributor requests to contribute to a project and if is accepted will be able to make changes based on the rules the owner or manager of the project has set. Which will be available later on.